src/Core/RenderAPI/EspPipelineBuilder.hh

```diff
@@ -0,0 +1,377 @@
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_6x5_SFLOAT_BLOCK = 1000066003,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_6x6_SFLOAT_BLOCK = 1000066004,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_8x5_SFLOAT_BLOCK = 1000066005,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_8x6_SFLOAT_BLOCK = 1000066006,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_8x8_SFLOAT_BLOCK = 1000066007,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_10x5_SFLOAT_BLOCK = 1000066008,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_10x6_SFLOAT_BLOCK = 1000066009,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_10x8_SFLOAT_BLOCK = 1000066010,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_10x10_SFLOAT_BLOCK = 1000066011,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_12x10_SFLOAT_BLOCK = 1000066012,
+    // Provided by VK_VERSION_1_3
+    ESP_FORMAT_ASTC_12x12_SFLOAT_BLOCK = 1000066013,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC1_2BPP_UNORM_BLOCK_IMG = 1000054000,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC1_4BPP_UNORM_BLOCK_IMG = 1000054001,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC2_2BPP_UNORM_BLOCK_IMG = 1000054002,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC2_4BPP_UNORM_BLOCK_IMG = 1000054003,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC1_2BPP_SRGB_BLOCK_IMG = 1000054004,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC1_4BPP_SRGB_BLOCK_IMG = 1000054005,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC2_2BPP_SRGB_BLOCK_IMG = 1000054006,
+    // Provided by VK_IMG_format_pvrtc
+    ESP_FORMAT_PVRTC2_4BPP_SRGB_BLOCK_IMG = 1000054007,
+    // Provided by VK_NV_optical_flow
+    ESP_FORMAT_R16G16_S10_5_NV = 1000464000,
+    // Provided by VK_KHR_maintenance5
+    ESP_FORMAT_A1B5G5R5_UNORM_PACK16_KHR = 1000470000,
+    // Provided by VK_KHR_maintenance5
+    ESP_FORMAT_A8_UNORM_KHR = 1000470001,
+  };
+
+  struct EspVertexAttribute
+  {
+    uint32_t m_location;
+    EspAttrFormat m_format;
+    uint32_t m_offset;
+
+    EspVertexAttribute(uint32_t location, EspAttrFormat format, uint32_t offset);
+  };
+
+  struct EspVertexLayout
+  {
+    uint32_t m_size;
+    uint32_t m_binding;
+    std::vector<EspVertexAttribute> m_attrs;
+
+    EspVertexLayout(uint32_t size, uint32_t binding, std::vector<EspVertexAttribute> attr);
+  };
+
+  class EspPipelineBuilder
+  {
+   public:
+    virtual ~EspPipelineBuilder() {}
+
+    virtual void set_shaders(std::string path_vertex, std::string path_fragment) = 0;
+    virtual void set_vertex_shader(std::string path_vertex)                      = 0;
+    virtual void set_fragment_shader(std::string path_fragment)                  = 0;
+
+    virtual void set_vertex_layouts(std::vector<EspVertexLayout> vertex_layouts)          = 0;
+    virtual void set_pipeline_layout(std::unique_ptr<EspUniformMetaData> pipeline_layout) = 0;
```

Comment:
I don't really like this naming convention here. Why is this variable called "pipeline_layout" if it is not a pipeline layout? It's data needed to create one, I agree, but imo it should be named simply sth like "uniform_meta_data" to avoid some misunderstandings.